src/Integration.Vsix/CFamily/VcxProject/IFileConfig.cs

```diff
@@ -0,0 +1,56 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject
+{
+    internal interface IFileConfig
```

Comment:
Created an interface for ease of mocking in VcxRequestFactoryTests; otherwise it was difficult to check that certain properties (namely AbsoluteFilePath) were indeed checked.

Reply:
Let's start with them separate and see what happens if/when a new field gets added.